tests/functional/src/main/java/org/mule/tck/junit4/ExtensionsFunctionalTestCase.java

```diff
@@ -0,0 +1,283 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.tck.junit4;
+
+import org.mule.DefaultMuleContext;
+import org.mule.api.MuleContext;
+import org.mule.api.config.ConfigurationBuilder;
+import org.mule.api.context.MuleContextAware;
+import org.mule.api.registry.SPIServiceRegistry;
+import org.mule.api.registry.ServiceRegistry;
+import org.mule.config.builders.AbstractConfigurationBuilder;
+import org.mule.extension.ExtensionManager;
+import org.mule.extension.introspection.Describer;
+import org.mule.extension.introspection.Extension;
+import org.mule.extension.introspection.ExtensionFactory;
+import org.mule.extension.resources.GenerableResource;
+import org.mule.extension.resources.ResourcesGenerator;
+import org.mule.extension.resources.spi.GenerableResourceContributor;
+import org.mule.module.extension.internal.introspection.AnnotationsBasedDescriber;
+import org.mule.module.extension.internal.introspection.DefaultExtensionFactory;
+import org.mule.module.extension.internal.manager.DefaultExtensionManager;
+import org.mule.module.extension.internal.resources.AbstractResourcesGenerator;
+import org.mule.util.ArrayUtils;
+import org.mule.util.IOUtils;
+
+import com.google.common.collect.ImmutableList;
+
+import java.io.File;
+import java.io.IOException;
+import java.lang.reflect.Method;
+import java.net.URL;
+import java.util.List;
+
+import org.apache.commons.io.FileUtils;
+
+/**
+ * Base test class for {@link FunctionalTestCase}s
+ * that make use of components generated through the extensions API.
+ * <p/>
+ * The value added by this class in comparison to a traditional
+ * {@link FunctionalTestCase} is that before creating
+ * the {@link MuleContext}, it creates a {@link ExtensionManager}
+ * and automatically discovers extensions by delegating on
+ * {@link ExtensionManager#discoverExtensions(ClassLoader)}.
```

Comment:
In general, users will be testing a single extension. Would not be better to provide a way to discover the extension overriding a method in order to return the extension under test? (to avoid the performance degradation involved in a search)